### PR TITLE
codeowners update: add codeowners for api machinery related files

### PR DESCRIPTION
**Pull request by yharish991**
Signed-off-by: Harish Yayi <yharish991@gmail.com>

### What this PR does / why we need it
This PR adds @vmware-tanzu/tanzu-framework-api-machinery-owners as the codeowners for API machinery related files

### Which issue(s) this PR fixes
Fixes https://github.com/vmware-tanzu/tanzu-framework/issues/1180

### Release note

```release-note
NONE
```

### PR Checklist

- [x] Squash the commits into one or a small number of logical commits
      
- [x] Use good commit [messages](https://github.com/vmware-tanzu/tanzu-framework/blob/main/CONTRIBUTING.md)
- [x] Ensure PR contains terms all contributors can understand and links all contributors can access

### CODE REVIEW

File: CODEOWNERS
Comment (rajathagasthya) [inline CODEOWNERS:29]: Should we leave this to `@vmware-tanzu/tanzu-framework-reviewers` or add other teams (at least the addons team) since this is common for everyone?

File: CODEOWNERS
Comment (yharish991) [reply CODEOWNERS:21]: ya i thought about it, but didn't want to add each and every file that belongs to us in the CODEOWNERS file, how about adding `config` and `run` dirs to ours?

File: CODEOWNERS
Comment (yharish991) [reply CODEOWNERS:29]: since these packages would be added/generated using the tooling we built I thought we should keep an eye on that dir for while at least until the process of adding the packages become solid.. we can continue to keep an eye or just add featuregates and capabilities dirs under our umbrella once the process becomes good enough.

Comment (yharish991) [reply CODEOWNERS:29]: since these packages would be added/generated using the tooling we built I thought we should keep an eye on that dir for while at least until the process of adding the packages become solid.. we can continue to keep an eye or just add featuregates and capabilities dirs under our umbrella once the process becomes good enough.

**Review by rajathagasthya — APPROVED** [review]

**Review by jmoroski — APPROVED** [review]

**Pull request merged**
